src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -90,24 +182,24 @@ .maxstack  2
   IL_0010:  ldarg.0
   IL_0011:  ldfld      ""int C._x""
   IL_0016:  call       ""void System.Console.WriteLine(int)""
-  IL_001b:  ldloca.s   V_0
-  IL_001d:  call       ""void C.<M>g__L12_0(ref C.<>c__DisplayClass2_0)""
-  IL_0022:  ldarg.0
-  IL_0023:  ldfld      ""int C._x""
-  IL_0028:  call       ""void System.Console.WriteLine(int)""
-  IL_002d:  ret
+  IL_001b:  ldarg.0
```

Comment:
It looks like we're storing `this` in both `C.<>c__DisplayClass2_0.<>4__this` as well as the receiver on `C.<M>g__L12_0(ref C.<>c__DisplayClass2_0)`. Is that intended?

(It looks like `C.<M>g__L12_0(ref C.<>c__DisplayClass2_0)` never touches the captured `C C.<>c__DisplayClass2_1.<>4__this`)